tests/fsharp/single-test.fs

```diff
@@ -161,136 +150,101 @@ let generateProjectArtifacts (pc:ProjectConfiguration) targetFramework configura
+         .Replace("$(LOADSOURCEITEMS)", computeSourceItems pc true true CompileItem.LoadSource pc.LoadSources)
+         .Replace("$(USESOURCEITEMS)", computeSourceItems pc true true CompileItem.UseSource pc.UseSources)
+         .Replace("$(DIRECTORYBUILDLOCATION)", Path.GetFullPath(__SOURCE_DIRECTORY__))
+         .Replace("$(OUTPUTTYPE)", outputType)
+         .Replace("$(OPTIMIZE)", optimize)
+         .Replace("$(DEBUG)", debug)
+         .Replace("$(TARGETFRAMEWORK)", targetFramework)
+         .Replace("$(RestoreFromArtifactsPath)", Path.GetFullPath(__SOURCE_DIRECTORY__) + "/../../artifacts/packages/" + configuration)
+
+    projBody
+
+let potentialSources = ["testlib.fsi";"testlib.fs";"test.mli";"test.ml";"test.fsi";"test.fs";"test2.fsi";"test2.fs";"test.fsx";"test2.fsx"]
+
+let executeSingleTestBuildAndRun cfg outputType testCompilerVersion targetFramework optimize extraRef =
     let loadSources = []
     let useSources = []
-    let extraSources = ["testlib.fsi";"testlib.fs";"test.mli";"test.ml";"test.fsi";"test.fs";"test2.fsi";"test2.fs";"test.fsx";"test2.fsx"]
+    let extraSources = potentialSources |> List.filter (fileExists cfg)
     let utilitySources = [__SOURCE_DIRECTORY__  ++ "coreclr_utilities.fs"]
-    let referenceItems =  if String.IsNullOrEmpty(copyFiles) then [] else [copyFiles]
-    let framework = "netcoreapp2.0"
-
-    let executeSingleTestBuildAndRun outputType compilerType targetFramework optimize =
-        let mutable result = false
-        let directory = Path.Combine(Path.GetTempPath(), Path.GetRandomFileName() )
-        let pc = {
-            OutputType = outputType
-            Framework = framework
-            SourceDirectory = cfg.Directory
-            SourceItems = sources
-            ExtraSourceItems = extraSources
-            UtilitySourceItems = utilitySources
-            ReferenceItems = referenceItems
-            LoadSources = loadSources
-            UseSources = useSources
-            Optimize = optimize
-        }
-
-        let targetsBody = generateTargets
-        let overridesBody = generateOverrides
-        let targetsFileName = Path.Combine(directory, "Directory.Build.targets")
-        let propsFileName = Path.Combine(directory, "Directory.Build.props")
-        let overridesFileName = Path.Combine(directory, "Directory.Overrides.targets")
-        let projectFileName = Path.Combine(directory, Path.GetRandomFileName() + ".fsproj")
-        try
-            // Clean up directory
-            Directory.CreateDirectory(directory) |> ignore
-            copyFilesToDest cfg.Directory directory
-            try File.Delete(Path.Combine(directory, "FSharp.Core.dll")) with _ -> ()
-            emitFile targetsFileName targetsBody
-            emitFile overridesFileName overridesBody
-            if outputType = OutputType.Exe then
-                let executeFsc testCompilerVersion targetFramework =
```

Comment:
@KevinRansom I removed these little functions. Generally 

1. Inner functions should be avoided
2. especially if only every called once
3. even more so if only called once immediately after the function is defined.